hmda-data-publisher/src/test/scala/hmda/publisher/helper/QuarterTimeBarrierTest.scala

```diff
@@ -31,20 +37,22 @@ class QuarterTimeBarrierTest extends FreeSpec {
       test(endDate.plusDays(8), quarter, shouldRun = false)
       test(endDate.plusDays(100), quarter, shouldRun = false)
     } else {
+      QuarterTimeBarrier.setRulesConfigWithActualYear(actualYear)
       val endDate = QuarterTimeBarrier.getEndDateForQuarter(quarter)
-      test(endDate.minusDays(100), quarter, shouldRun = false)
-      test(endDate, quarter, shouldRun = true)
-      test(endDate.plusDays(6), quarter, shouldRun = true)
-      test(endDate.plusDays(7), quarter, shouldRun = true)
-      test(endDate.plusDays(8), quarter, shouldRun = false)
-      test(endDate.plusDays(100), quarter, shouldRun = false)
+      test(endDate.minusDays(100), quarter, shouldRun = false, actualYear)
+      test(endDate, quarter, shouldRun = true, actualYear)
+      test(endDate.plusDays(6), quarter, shouldRun = true, actualYear)
+      test(endDate.plusDays(7), quarter, shouldRun = true, actualYear)
+      test(endDate.plusDays(8), quarter, shouldRun = false, actualYear)
+      test(endDate.plusDays(100), quarter, shouldRun = false, actualYear)
     }
   }
 
-  def test(now: LocalDate, quarter: Period.Quarter, shouldRun: Boolean) = {
+  def test(now: LocalDate, quarter: Period.Quarter, shouldRun: Boolean, actualYear: Int = 0) = {
     val zoneId = ZoneId.systemDefault()
     val clock = Clock.fixed(now.atTime(12, 0).toInstant(zoneId.getRules.getOffset(Instant.now())), zoneId)
     val timeBarrier = new QuarterTimeBarrier(clock)
+    timeBarrier.setRulesConfigWithActualYear(actualYear)
```

Comment:
We should only need to call this method once, either here or in the `testQuarter` method. Inside `testQuarter` is probably better imo

Reply:
Nevermind, you are correct that we need it twice: once for getting the quarter's end date in `testQuarter` and again when we check if the code should have run with the new timeBarrier object here in` test`